Objects/regions.c

```diff
@@ -643,9 +675,294 @@ PyObject* _Py_MakeImmutable(PyObject* obj)
 
     stack_free(frontier);
 
-    _Py_VPYDBGPRINT(obj);
-    _Py_VPYDBG(" region: %lu rc: %ld \n", Py_REGION(obj), Py_REFCNT(obj));
-    _Py_VPYDBG("<< makeimmutable complete\n\n");
 
-    return obj;
-}
\ No newline at end of file
+    Py_RETURN_NONE;
```

Comment:
> Why is this returning None now?

Both returning the object and none are valid options. I think this is more a question of usability. Having `makeimmutable()` return the object allows the inline usage, like `r.x = makeimmutable(x)` and similar. However, it also requires us to handle reference counts. If we return the object we have to `Py_INCREF()` it in `_Py_MakeImmutable`. Calling C code will then always need to call `Py_DECREF()`. Since there wasn't a usage yet that used `makeimmutable()` ability to be inline, I just picked the easier option, thus returning `None`.

> If we're going to return None, we should do that everywhere (e.g. Line 595)

Good catch the `return NULL` was added before this change returning `None`

Reply:
Good point re: ref counting. Refcount is obviously not an issue at the moment (all immutable objects also become immortal) but eventually we do want them to be GCed.